### ISSUE TITLE
Can't apply a resource more than two times

### ISSUE THREAD

**Issue by gintautassulskus** [coreos/prometheus-operator#699]
Hi,

I am not entirely sure if this issue is specific to pormetheus or kubernetes. Thought to leave it for you to judge.  

**The issue**

`kubectl` throws an error when a `ServiceMonitor` resource is applied for the _third_ time. Might be the same with other custom resource definitions as well. 

The following displays the result of an attempt to apply the contents of resources.yaml (defined later in the post). Initially, the resource in question is not present in the cluster.
```
$ kubectl apply -f ./resources.yaml
servicemonitor "kube-state-metrics" created
$ kubectl apply -f ./resources.yaml
servicemonitor "kube-state-metrics" configured
$ kubectl apply -f ./resources.yaml
The ServiceMonitor "kube-state-metrics" is invalid: apiVersion: Invalid value: "ServiceMonitor": must be monitoring.coreos.com/v1
```

**What did you expect to see?**
servicemonitor "kube-state-metrics" configured

**What did you see instead? Under which circumstances?**

The ServiceMonitor "kube-state-metrics" is invalid: apiVersion: Invalid value: "ServiceMonitor": must be monitoring.coreos.com/v1

**Environment**

* Kubernetes version information:

Client Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.1", GitCommit:"f38e43b221d08850172a9a4ea785a86a3ffa3b3a", GitTreeState:"clean", BuildDate:"2017-10-11T23:27:35Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.1+coreos.0", GitCommit:"59359d9fdce74738ac9a672d2f31e9a346c5cece", GitTreeState:"clean", BuildDate:"2017-10-12T21:53:13Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}

* Kubernetes cluster kind: 

hyperkube 1.8.1

* Manifests:

resources.yaml:
```
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: kube-state-metrics
  labels:
    k8s-app: kube-state-metrics
  namespace: kube-system
spec:
  jobLabel: k8s-app
  selector:
    matchLabels:
      k8s-app: kube-state-metrics
  namespaceSelector:
    matchNames:
    - monitoring
  endpoints:
  - port: http-metrics
    interval: 30s
    honorLabels: true
```

* Prometheus Operator Logs:

Nothing is logged, that is why I think this might be kubernetes at fault, not prometheus.

**Comment by brancz** [comment]
Very odd. Yes this seems to be a problem with Kubernetes, thanks for reporting, we will investigate.

**Comment by yoniranjan** [comment]
We're having the same issue. If applied for two or three times, the following errors surface up:

The Alertmanager "main" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
The ServiceMonitor "alertmanager" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
The ServiceMonitor "kube-apiserver" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
The ServiceMonitor "kube-controller-manager" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
The ServiceMonitor "kube-dns" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
The ServiceMonitor "kube-scheduler" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
The ServiceMonitor "kube-state-metrics" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
The ServiceMonitor "kubelet" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
The ServiceMonitor "node-exporter" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
The ServiceMonitor "prometheus-operator" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
The ServiceMonitor "prometheus" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1

monitoring.coreos.com/__internal does not show up in any of the yml files.

**Comment by r0bj** [comment]
The same situation with `ServiceMonitor` definition:
```
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: kube-state-metrics
  labels:
    k8s-app: kube-state-metrics
  namespace: monitoring
spec:
  jobLabel: k8s-app
  selector:
    matchLabels:
      k8s-app: kube-state-metrics
  namespaceSelector:
    matchNames:
    - kube-system
  endpoints:
  - port: http-metrics
    interval: 30s
    honorLabels: true
```

```
# kubectl apply -f prometheus-k8s-service-monitor.yaml
servicemonitor "kube-state-metrics" configured
# kubectl apply -f prometheus-k8s-service-monitor.yaml
The ServiceMonitor "kube-state-metrics" is invalid: apiVersion: Invalid value: "monitoring.coreos.com/__internal": must be monitoring.coreos.com/v1
```

**Comment by brancz** [comment]
Yes, this seems to be a problem with Kubernetes itself. I haven't been able to get to the ground of what's happening. I'm imaging a problem with the merge-patch functionality with custom resources.

**Comment by swestcott** [comment]
Just cross referencing the upstream issue for anyone else who runs into this. The fix is expected in Kubernetes 1.8.5.

https://github.com/kubernetes/kubernetes/issues/53379

**Comment by brancz** [comment]
This issue was since fixed upstream :tada: . Closing here.
